executor.go

```diff
@@ -1358,65 +1444,156 @@ func (e *executor) executePercentile(ctx context.Context, qcx *Qcx, index string
 	if err != nil {
 		return ValCount{}, errors.Wrap(err, "executing Max call for Percentile")
 	}
-	// set up reusables
+
+	if desiredGreater == 0 {
+		if maxVal.DecimalVal != nil {
+			maxVal.FloatVal = maxVal.DecimalVal.Float64()
+		}
+		return maxVal, nil
+	}
+
+	// the logic here is basically identical whether we're doing a decimal field
+	// or an integer field, but the actual code used to compare maximum and minimum
+	// values, or extract values from valCount objects, differs.
+	// So we set up generic functions which will produce the right values.
+	var averageMinMax func() interface{}
+	var minLessthanMax func() bool
+	var maxValueUnder func(interface{})
+	var minValueOver func(interface{})
+
+	if field.options.Type == FieldTypeDecimal {
+		minPtr := minVal.DecimalVal
+		maxPtr := maxVal.DecimalVal
+		if minPtr == nil || maxPtr == nil {
+			return nil, fmt.Errorf("unexpectedly nil min/max values")
```

Comment:
for diagnosing things I think I'd prefer if nil minPtr and nil maxPtr gave separate error messages.

Reply:
that's a good call and i wish i'd read this BEFORE i propagated changes to dax.